### PR TITLE
A0-1259: Add metadata to PSP22 contracts

### PR DESCRIPTION
**Pull request by obrok**
# Description

Implements https://cardinal-cryptography.atlassian.net/browse/A0-1259

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)

# Checklist:

- I have made corresponding changes to the existing documentation

### CODE REVIEW

**Comment by obrok** [comment]
@fbielejec there is no third token in `deploy.sh` as mentioned in jira - is that one not yet merged?

**Comment by fbielejec** [comment]
> @fbielejec there is no third token in `deploy.sh` as mentioned in jira - is that one not yet merged?

In progress in another PR

**Review by kostekIV — APPROVED** [review]
Looks great :)

File: contracts/scripts/deploy.sh [hunk 1]
```diff
@@ -8,13 +8,15 @@ function instrument_game_token {
 
   local  __resultvar=$1
   local contract_name=$2
-  local salt=$3
+  local token_name=\"$3\"
```
[the remark this answers is not in the document]
Comment (fbielejec) [reply contracts/scripts/deploy.sh:11]: I suppose otherwise cli doesn't recognize it's a String

**Review by fbielejec — APPROVED** [review]

File: contracts/game_token/lib.rs [hunk 2]
```diff
@@ -11,24 +11,32 @@ pub mod game_token {
...
     use ink_storage::traits::SpreadAllocate;
-    use openbrush::{contracts::psp22::*, traits::Storage};
+    use openbrush::{
+        contracts::psp22::{extensions::metadata::*, Internal},
+        traits::Storage,
+    };
 
     pub const BALANCE_OF_SELECTOR: [u8; 4] = [0x65, 0x68, 0x38, 0x2f];
     pub const TRANSFER_SELECTOR: [u8; 4] = [0xdb, 0x20, 0xf9, 0xf5];
+    pub const TOKEN_DECIMALS: u8 = 18;
```
Comment (fbielejec) [inline contracts/game_token/lib.rs:23]: this does not need to be a const, just hardcode in constructor

File: contracts/game_token/lib.rs [hunk 3]
```diff
@@ -116,6 +127,9 @@ pub mod game_token {
 
             match role_check {
                 Ok(_) => ink_lang::codegen::initialize_contract(|instance: &mut GameToken| {
+                    instance.metadata.name = Some(name);
+                    instance.metadata.symbol = Some(symbol);
+                    instance.metadata.decimals = TOKEN_DECIMALS;
```
Comment (fbielejec) [inline contracts/game_token/lib.rs:132]:
```suggestion
                    instance.metadata.decimals = 18;
```

File: contracts/scripts/deploy.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (obrok) [reply contracts/scripts/deploy.sh:11]: Indeed

**Pull request merged**